language-extensions/dotnet-core-CSharp/src/managed/CSharpSession.cs

```diff
@@ -192,7 +192,9 @@ public void Execute(
             if(_outputDataSet.CSharpDataFrame != null)
             {
                 _outputDataSet.ColumnsNumber = (ushort)_outputDataSet.CSharpDataFrame.Columns.Count;
-                _outputDataSet.ExtractColumns(_outputDataSet.CSharpDataFrame);
+                // Pass input column metadata to preserve data types (e.g., nvarchar vs varchar).
+                //
+                _outputDataSet.ExtractColumns(_outputDataSet.CSharpDataFrame, _inputDataSet.Columns);
```

Comment:
why would information about data type of the inputDataSet be useful to the outputDataSet? Might be useful only in case OutputDataSet = InputDataSet for example in Passthrough queries.